easypy/threadtree.py

```diff
@@ -20,17 +22,72 @@
 
 
 _REGISTER_GREENLETS = False
+_FRAME_SNAPSHOTS_REGISTRY = defaultdict(dict)
+
 _orig_start_new_thread = _thread.start_new_thread
 
 
+class FCode(NamedTuple):
+    co_name: str
+    co_filename: str
+    co_firstlineno: Optional[int]
+    co_names: Tuple[str, ...] = tuple("_frame_snapshot__co_names_skipped")
+
+class FrameSnapshot(NamedTuple):
+    f_lineno: int
+    f_lasti: int
+    f_code: FCode
+    f_thread_ident: int
+    f_globals: Dict[str, Any]
+    f_back: Optional["FrameSnapshot"]
+    f_locals: Dict = {"_frame_snapshot__locals_skipped": None}
+
+    def __repr__(self) -> str:
+        return (
+            f"FrameSnapshot(thread {self.f_thread_ident}, file '{self.f_code.co_filename}', "
+            f"line {self.f_lineno}, code {self.f_code.co_name})"
+        )
+
+
+def create_frame_snapshot(frame=None, thread=None) -> FrameSnapshot:
+    if not frame:
+        frame = sys._getframe(0)
+
+    if not thread:
+        thread = threading.current_thread()
+
+    snapshot = FrameSnapshot(
+        f_lineno=frame.f_lineno,
+        f_lasti=frame.f_lasti,
+        f_code=FCode(
+            co_name=frame.f_code.co_name,
+            co_filename=frame.f_code.co_filename,
+            co_firstlineno=getattr(frame.f_code, "f_firstlineno", None),
+        ),
+        f_globals={
+            "__name__": frame.f_globals.get("__name__"),
+            "_frame_snapshot__globals_skipped": None,
+        },
+        f_thread_ident=thread.ident,
+        f_back=create_frame_snapshot(frame.f_back, thread) if frame.f_back else None,
+    )
+    return snapshot
+
+
 def start_new_thread(target, *args, **kwargs):
     """
-    A wrapper for the built in 'start_new_thread' used to capture the parent of each new thread.
+    A wrapper for the built-in 'start_new_thread' used to capture the parent of each new thread.
     """
-    parent_uuid = get_thread_uuid()
+    parent_thread = threading.current_thread()
+    parent_uuid = get_thread_uuid(parent_thread)
+    parent_frame = sys._getframe(0)
 
     def wrapper(*args, **kwargs):
         thread = threading.current_thread()
+
+        _FRAME_SNAPSHOTS_REGISTRY[parent_thread.ident][thread.ident] = (
```

Comment:
this would work as well, saving you from needing a `defaultdict`
```suggestion
        _FRAME_SNAPSHOTS_REGISTRY[parent_thread.ident, thread.ident] = (
```
(you're not really leveraging the nested, tree structure anyway)